module-2/backend/week_four.md

```diff
@@ -10,14 +10,44 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+  - A cookie is a small file that lives on the client end. It is generated in the response cycle. It helps create the impression of "state" by passing back relevant information in the request process. This can help the server remember who we are and what we were up to last time our devices talked to eachother.
+  
 * What’s the difference between a session and a cookie?
+    - A cookie is is stored in the server/locally.
+    - A session is stored in the browser.
+
 * What’s a flash and when do you want to use flashes?
+  - A flash is a way to let your user know what is happening. This allows them to know if they completed a task satisfactorally or something failed. It's a way to conditionally send messages to your user when "stuff" happens.
+  
 * Why do people say “HTTP is stateless”?
+  - Becasue of the request response cycle. You are not "live and tapped into the server". You are making constant requests and getting constant responses.
+  
 * What’s authentication? Explain.
+  -  Authentication is checking that somebody is who they say you are.
+  
 * What’s the difference between authentication and authorization?
+  - Once again, Authentication is checking that somebody is who they say you are.
+  - Authorization make sure you are where you are supposed to be and not in places you don't belong.
+
 * What’s a before filter?
+  - It checks for authorization level allowing a user to do something. Still mildly fuzzy on this.
+
 * How do we keep track of a user once they’ve logged in?
+  - Through a session
+
 * When do you want to namespace a resource? When do you want to nest a resource? What's the differences between those two approaches?
+  - Namespacing a resource can allow you to also namespace the controller for things like Admin functions.
+  - Nesting a resource allows things, in once case, to belong to another thing. Like '/companies/1/jobs/3' for example. This can give a better representation of the database relationship as well.
+  
 * At a high level, what tools can you use to implement authorization? How would you use them?
+  - User Roles.
+  - Namespacing
+
 * What's an enum, and what advantages does it offer? What data type needs to be in your database to use an enum? Where do you declare an enum?
+  - It allows us to set different levels of access to a user based on an array of roles that corelate with an index positions (0,1,etc) which would allow different levels of authorization.
```

Comment:
Doesn't have to be levels of access. Can be used any time we want to translate numbers in our database to words at the model level. Also gives us nice `?` and `!` methods to make changes to instances of our models.